backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/deliveryplanning/DeliveryPlanningRepository.java

```diff
@@ -125,4 +125,20 @@ public boolean isOtherDeliveryPlanningsExistForOrderLine(@NonNull final OrderLin
 				.create()
 				.anyMatch();
 	}
+
+	public void deleteForReceiptSchedule(@NonNull final ReceiptScheduleId receiptScheduleId)
+	{
+		queryBL.createQueryBuilder(I_M_Delivery_Planning.class)
+				.addEqualsFilter(I_M_Delivery_Planning.COLUMNNAME_M_ReceiptSchedule_ID, receiptScheduleId)
+				.create()
+				.deleteDirectly();
```

Comment:
because I want to skip the interceptor that forbids deletion on the last delivery planning per order line: https://github.com/metasfresh/metasfresh/blob/b45b7d7e2d451c322b9833973c99b6883cd55e84/backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/deliveryplanning/interceptor/M_Delivery_Planning.java#L45

Reply:
Fixed by using the org.adempiere.model.InterfaceWrapperHelper#isUIAction in de.metas.deliveryplanning.interceptor.M_Delivery_Planning#onDelete